### ISSUE TITLE
Cannot use 'in' operator to search for 'Symbol(StrapiCustomCoreController)' in undefined

### ISSUE THREAD

**Issue by alextrastero** [strapi/strapi#19135]
Hi, i'm trying to debug why this error appears when running strapi develop.

Error:
```
[2024-01-02 14:37:18.940] debug: ⛔️ Server wasn't able to start properly.
[2024-01-02 14:37:18.942] error: Cannot use 'in' operator to search for 'Symbol(StrapiCustomCoreController)' in undefined
TypeError: Cannot use 'in' operator to search for 'Symbol(StrapiCustomCoreController)' in undefined
    at isCustomController (/home/dev/gn/strapi-cms/node_modules/@strapi/strapi/dist/factories.js:80:35)
    at Array.filter (<anonymous>)
    at Strapi.sendStartupTelemetry (/home/dev/gn/strapi-cms/node_modules/@strapi/strapi/dist/Strapi.js:243:68)
    at Strapi.postListen (/home/dev/gn/strapi-cms/node_modules/@strapi/strapi/dist/Strapi.js:266:10)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Server.onListen (/home/dev/gn/strapi-cms/node_modules/@strapi/strapi/dist/Strapi.js:276:11)
```

Controller:
```
export default {}
```

package.json:
```
  "scripts": {
    "lint": "eslint .",
    "develop": "strapi develop",
    "start": "strapi start",
    "build": "strapi build",
    "strapi": "strapi"
  },
  "dependencies": {
    "@google-cloud/pubsub": "^4.0.3",
    "@strapi/plugin-users-permissions": "^4.12.4",
    "@strapi/strapi": "^4.16.2",
    "@strapi/utils": "^4.9.2",
    "axios": "^1.4.0",
    "lodash": "^4.17.21",
    "monaco-editor-webpack-plugin": "^7.1.0",
    "pg": "8.8.0",
    "react": "^18.2.0",
    "react-dom": "^18.2.0",
    "react-router-dom": "^5.3.4",
    "strapi-plugin-import-export-entries": "^1.21.0",
    "strapi-provider-search-algolia": "file:providers/strapi-provider-search-algolia",
    "styled-components": "^5.3.11"
  },

```

**Comment by alextrastero** [comment]
I know this is not the solution but in order to make the admin work I edited:
`node_modules/@strapi/strapi/dist/factories.js`

```js
const isCustomController = (controller) => {
  if (controller) {
    return symbols.CustomController in controller;
  } else {
    console.error('controller undefined!');
  }

  return false;
};
```

**Comment by alextrastero** [comment]
I guess my question is, in what possible scenario would this `controller` be undefined when calling `isCustomController` ?

**Comment by alextrastero** [comment]
I'm going to post the solution here in case anyone finds it useful.

**Issue:** strapi running into undefined controller

This was caused because within the controller folder I had a constants file.

```
├── metric                                                                                                                                                                                                                            
│   ├── content-types              
│   │   └── metric          
│   │       ├── cookie.ts          
│   │       ├── lifecycles.ts
│   │       ├── schema.json        
│   │       └── utils.ts   
│   ├── controllers  
│   │   └── metric.ts         <- no other files can live in here
│   ├── EVENTS.md          
│   ├── routes        
│   │   └── metric.ts            
│   ├── services      
│   │   └── metric.ts            
│   └── utils        
│       ├── constants.ts                 
│       ├── index.ts       
│       └── types.ts 
```

**Comment by Convly** [comment]
Hey, thanks for reporting this issue!

I was looking at how to reproduce but couldn't figure out how you ended up with an `undefined` controller :exploding_head: 

I'm gonna propose a fix but I don't think `isCustomController` is at fault here. I'm gonna filter out undefined controllers from being checked higher up in the call chain.

Again, thanks for creating the bug report, appreciated it :slightly_smiling_face:

**Comment by alextrastero** [comment]
Thanks @Convly, it is an edge case indeed.
We updated from `4.9.2` to `4.15.5` and this started showing, we housed within `api/[name]/controller/` folder other modules like constants and utils. And these were being identified as undefined controllers

**Comment by Convly** [comment]
> other modules like constants and utils

Yeah, Strapi will load everything it finds in your controllers' directories and try to interpret them as controller files.

We've merged the fixed and it should be available in `4.17.0` or `1.16.3` depending on which one gets released first. I'm gonna close this as the issue should be resolved in the next release. Please feel free to reopen it if it persists.

Have a nice day (and year)!

**Comment by alextrastero** [comment]
🙏🏼 likewise
